### PR TITLE
Added a button and code to clear gcode.

### PR DESCRIPTION
**Pull request by thormj**
I wanted to be able to clear the g-code buffer to clean up the screen.
Plus, this is a way for me to see how to use GitHub...

### CODE REVIEW

**Comment by blurfl** [comment]
This does what it says on the box. It leaves the gcode file selection in groundcontrol.ini alone so that on next launch the previously selected gcode file is back and the location is preserved. 
 You've mastered the GitHub thing on the first go!
 I probably won't use this option very often, though. I've got an empty gcode file with only the setup lines (G90, G20) that I use as a starting point, it accomplishes much the same thing.

**Comment by BarbourSmith** [comment]
Fantastic! Excellent mastery of GitHub!

For anyone following along here's what the button looks like:

image

How do we feel about moving the button over into one of the other less crowded columns? Maybe replacing that "zero z" button that has been moved into the z-axis popup?

To make changes, just make a new commit to the same branch as you did before (ClearGCode) and it will show up here automatically 👍 👍

**Comment by thormj** [comment]
I thought of that, but I kinda like how each column is dedicated to something "View" (everything dealing with code) (maybe should be "Code" instead),  Run (if you hit these buttons something happens).

OTOH, I suppose you could put a button on the file picker to "clear".

I was also trying to pass in python main.py -c:data:GCodeFilename:"" to see if it would blank a file, but evidently I don't understand command line parameters much...

**Comment by blurfl** [comment]
I've had success with a command line argument to groundcontrol in the past, but can't find the code right now.
Take a look at this [guy's example](https://gist.github.com/jegger/28d3e295ae690e2fa2ac) to passing a command line value into kivy. I started from there, I think.

File: UIElements/gcodeCanvas.py [hunk 1]
```diff
@@ -156,8 +156,7 @@ def reloadGcode(self, *args):
                     else:
                         self.data.zMoves.append(index)
         except:
-            if filename is not "":
```
Comment (BarbourSmith) [inline UIElements/gcodeCanvas.py]: I think we added this to fix issue #520 . Sorry to nitpick!

**Comment by blurfl** [comment]
That patch for issue #520  isn't needed with the code brought by this PR. This handles the missing file by clearing the gcode list.

**Comment by BarbourSmith** [comment]
Great catch!

**Comment by BarbourSmith** [comment]
Sorry for missing that

**Comment by BarbourSmith** [comment]
In that case it looks great to me! 👍 👍

**Pull request merged**